### PR TITLE
feat(file-helpers): add formatFileSize function to format bytes into human-readable units

### PR DESCRIPTION
**Pull request by filobater**
### Short Summary

- Implemented a new utility function to format bytes into human-readable units
- Added tests to validate the functionality, including zero and null input.
- Updated README with usage examples for the new function.
- Updated the test of the delay to use fake timers from jest and this approach does not depend on the OS type 
- Added new package "rimraf" to handle building the project on windows 

## Summary by CodeRabbit

* **New Features**
  * Added a file-size formatter that converts bytes into human-readable strings with automatic unit selection (B, KB, MB, GB, TB). Bytes show no decimals; larger units show one decimal. Returns "0 B" for 0 or null.

* **Tests**
  * Added tests for B/KB/MB/GB and edge cases for the formatter. Made timer handling deterministic in an unrelated test.

* **Chores**
  * Updated build script to use a more robust clean command.

### CODE REVIEW

File: src/file.ts [hunk 1]
```diff
@@ -11,3 +11,26 @@ export const checkFileExtension = (filePath: string, validExtensions: string[]):
...
+ */
+export const formatFileSize = (bytes: number | null): string => {
+  if (!bytes || bytes === 0) return "0 B"
+
+  const units = ["B", "KB", "MB", "GB", "TB"]
+  let size = bytes
+  let unitIndex = 0
+
+  while (size >= 1024 && unitIndex < units.length - 1) {
+    size /= 1024
```
[the remark this answers is not in the document]
Comment (filobater) [reply src/file.ts]: addressed and filed a task on clickup

File: src/file.ts [hunk 2]
```diff
@@ -11,3 +11,23 @@ export const checkFileExtension = (filePath: string, validExtensions: string[]):
...
+export const formatFileSize = (bytes: number | null): string => {
+  if (!bytes || bytes === 0) return "0 B"
+
+  const units = ["B", "KB", "MB", "GB", "TB"]
+
+  const unitIndex = Math.floor(Math.log2(bytes) / 10)
+
+  const size = bytes / Math.pow(1024, unitIndex)
+
+  return `${size.toFixed(unitIndex === 0 ? 0 : 1)} ${units[unitIndex]}`
```
[the remark this answers is not in the document]
Comment (filobater) [reply src/file.ts]: addressed

File: src/file.ts [hunk 3]
```diff
@@ -11,3 +11,24 @@ export const checkFileExtension = (filePath: string, validExtensions: string[]):
...
+
+  const units = ["B", "KB", "MB", "GB", "TB"]
+
+  const unitIndex = Math.floor(Math.log2(bytes) / 10)
+  const clampedIndex = Math.min(unitIndex, units.length - 1)
+
+  const size = bytes / Math.pow(1024, clampedIndex)
+
+  return `${size.toFixed(clampedIndex === 0 ? 0 : 1)} ${units[clampedIndex]}`
+}
```
[the remark this answers is not in the document]
Comment (filobater) [reply src/file.ts]: addressed

File: __tests__/file.test.ts [hunk 4]
```diff
@@ -15,3 +15,29 @@ describe("checkFileExtension", () => {
...
+  })
+
+  it("returns '0 B' when the size is negative", () => {
+    expect(formatFileSize(-1)).toBe("0 B")
+  })
+
+  it("returns '0 B' when the input value is null (invalid size)", () => {
+    expect(formatFileSize(null)).toBe("0 B")
+  })
+})
```
[the remark this answers is not in the document]
Comment (filobater) [reply __tests__/file.test.ts]: addressed
